src/main/java/seedu/eylah/expensesplitter/logic/commands/AddItemCommand.java

```diff
@@ -30,22 +31,37 @@
     public static final String MESSAGE_SUCCESS = "The entry: \n  -> %1$s\nhas been added.";
 
     private Entry toBeAdded;
+    private ArrayList<Person> persons;
+    private Amount amount;
 
     /**
      * Creates an AddItemCommand to add the specified {@code Item}
      *
      * @param item Item to be added.
      * @param persons String array of persons to be added.
      */
-    public AddItemCommand(Item item, ArrayList<Person> persons) {
+    public AddItemCommand(Item item, ArrayList<Person> persons, Amount amount) {
         requireAllNonNull(item, persons);
+        this.persons = persons;
+        this.amount = amount;
         toBeAdded = new Entry(item, persons);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         model.addEntry(toBeAdded);
+        for (Person person : persons) {
+            if (model.hasPerson(person)) {
+                model.addAmount(person, amount);
+                // person.addAmount(amount); //not sure if necessary
+            } else {
+                // add person, then add amount to that person
+                model.addPerson(person);
+                model.addAmount(person, amount);
+                // person.addAmount(amount); //not sure if necessary
```

Comment:
I think this is kind of redundant seems the model already add the person with the amount, the person should not add amount again.

Reply:
addPerson() doesn't add amount. It simply creates the person in the personAmountBook

Reply:
Then how about `addAmount` ? Did this method add the person with the amount?

Reply:
`addAmount` simply adds the amount to the person